src/cluster/Backend.h

```diff
@@ -341,6 +342,11 @@ class Backend {
      */
     const std::string& NodeId() const { return node_id; }
 
+    void SetTelemetry(detail::TelemetryPtr new_telemetry) {
+        // Should we inject this through the constructor instead?
```

Comment:
The setter seems fine to me. The only downside is if the backend ever held a value telemetry before with the setter its data would get thrown away, but this seems expected and not something we'd typically do anyway.